tools/setup_services.py

```diff
@@ -19,6 +400,15 @@ def copy_supervisor_configs():
             ]
             services.update({s: pjoin(path, s) for s in path_services})
 
+    all_plugins_path = cfg.get("services.plugins_path", "./plugins")
+    for p in activated_plugins:
+        services.update({p: pjoin(all_plugins_path, p)})
+
+    # TODO (in a future PR): loop over all services, check if they are a git submodule or not
+    # if they are a submodule make sure they are initialized and updated
+    # this should be discussed, it does not seem necessary as soon as we have the
```

Comment:
I think user can just take care of submodule updates, then plugin can be specified as file

Reply:
The advantages I see w/ using the plugin system for submodules is version validation + in general giving some standard place in the config for params that these submodules should read from. But we don't need to do any branch/sha/version management. Any thoughts?

Reply:
I guess I'm thinking: the default services are essentially exactly the same as plugins, except they don't do version checking. So, a user can add submodules, check them out with their project, and augment the services path. We can update services to check skyportal version, if a `pyproject.toml` is available. That way we have one unified pipeline for both, and `plugins` just become a mechanism for checking out plugins from remote locations.